src/osemgrep/reporting/Summary_report.ml

```diff
@@ -37,21 +58,10 @@ let pp_summary ppf
   *)
   let out_limited =
     if respect_git_ignore then
-      (* # Each target could be a git repo, and we respect the git ignore
-         # of each target, so to be accurate with this print statement we
-         # need to check if any target is a git repo and not just the cwd
-         targets_not_in_git = 0
-         dir_targets = 0
-         for t in self.target_manager.targets:
-             if t.path.is_dir():
-                 dir_targets += 1
-                 try:
-                     t.files_from_git_ls()
-                 except (subprocess.SubprocessError, FileNotFoundError):
-                     targets_not_in_git += 1
-                     continue
-         if targets_not_in_git != dir_targets: *)
-      Some "Scan was limited to files tracked by git."
+      let any_git_repos = find_git_dir target_roots in
+      match any_git_repos with
+      | Some _ -> Some "Scan was limited to files tracked by git."
+      | _ -> None
```

Comment:
It doesn't matter here, but as a stylistic rule we prefer to write `| None -> None`. In general, unless there are truly too many cases to list, we try to list them out explicitly